### PR TITLE
Fix: Skeleton missing __iter__

### PR DESCRIPTION
**Pull request by patricia-causalens**
## Description and Motivation

When adding support for Skeleton on Dara, the graph was not json serialised correctly by fastapi. The reason being that `__iter__` was missing from the class.

## Additional Context

Needed for Dara to support `Skeleton` graphs on `CausalGraphViewer` component.

## How Has This Been Tested?

Tested that this fixed the issue I was having on Dara by manually applying this changes within the env

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor (moving code around, general refactoring improvements)
- [ ] Documentation (documentation)

## PR Checklist:

- [x] I have reviewed my own PR? (review the PR as you are reviewing someone else's PR)
- [x] I have implemented all requirements? (see JIRA, project documentation)
- [ ] I am affecting someone else's work? If so: I have added those people as reviewers?
- [ ] I have added comments to all the bits that are hard to follow
- [x] At a bare minimum, I tested this manually
- [ ] I have added unit test(s)
- [ ] Is this a bugfix? If so have I added a regression test?
- [x] Does this PR satisfy the [one PR, one concern](https://medium.com/@fagnerbrack/one-pull-request-one-concern-e84a27dfe9f1) rule?
- [ ] If needed, documentation has been added/updated?
- [x] I have updated the appropriate changelog with a line for my changes
- [ ] If this PR breaks reproducibility, have I added a note in the changelog explaining the break in reproducibility
      and its extent?

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## NEXT
+
+- Added `__iter__` to `cai_causal_graph.causal_graph.Skeleton`
```
Comment (ilya-causalens) [inline changelog.md:5]: missing period at the end.

File: cai_causal_graph/causal_graph.py [hunk 2]
```diff
@@ -94,6 +94,11 @@ def __ne__(self, other: object) -> bool:
         """Check if the skeleton is not equal to another skeleton."""
         return not (self == other)
 
+    def __iter__(self) -> Iterator:
+        """Return item,value tuples of the content of self.to_dict()."""
```
Comment (ilya-causalens) [inline cai_causal_graph/causal_graph.py:98]: wrap self.to_dict in "`" for better formatting.

**Review by ilya-causalens — APPROVED** [review]
Super minor things - looks good! :)

File: changelog.md (hunk 1 above)
Comment (patricia-causalens) [reply changelog.md]: Fixed

File: cai_causal_graph/causal_graph.py (hunk 2 above)
Comment (patricia-causalens) [reply cai_causal_graph/causal_graph.py]: Added on this and on the one for CausalGraph that was also missing :)

**Review by steven-causalens — APPROVED** [review]
LGTM

**Review by andrew-causalens — APPROVED** [review]

**Pull request merged**